sphinx/util/nodes.py

```diff
@@ -325,7 +325,10 @@ def traverse_translatable_index(
         yield node, entries
 
 
-def nested_parse_with_titles(state: RSTState, content: StringList, node: Node,
+_RSTContext = TypeVar("_RSTContext")
+
+
+def nested_parse_with_titles(state: RSTState[_RSTContext], content: StringList, node: Node,
```

Comment:
Why didn't the [same](https://github.com/sphinx-doc/sphinx/pull/12511#issuecomment-2209269245) `WARNING: py:class reference target not found: _RSTContext [ref.class]` failure occur here?

Reply:
Ah -- perhaps it's because of the `:param state:` line in the docstring of the former case; this function doesn't list parameters within its docstring?

However, it also seems that this function isn't listed in the generated documentation.  That could be another reason that the warning doesn't appear.